compiler/modules/nback.llvm/llvm.bal

```diff
@@ -257,8 +272,47 @@ public class Builder {
     // Corresponds to LLVMBuildCall
     // Returns () if there is no result i.e. function return type is void
     public function call(Function fn, Value[] args) returns Value? {
-        // XXX todo
-        return ();
+        if fn.paramCount() != args.length() {
+            panic error(string `Number of arguments is invalid for function ${fn.ref()}`);
+        }
+        BasicBlock bb = self.bb();
+        string[] argStringContent = [];
+        foreach int i in 0 ..< args.length() {
+            final Value arg = args[i];
+            if i > 0 {
+                argStringContent.push(",");
+            }
+            argStringContent.push(typeToString(arg.ty));
+            argStringContent.push(arg.operand);
+        }
+        if fn.returnsValue() {
+            string reg = bb.func.genReg();
+            string[] instruction = [];
+            instruction.push(reg);
+            instruction.push("=");
+            instruction.push("call");
```

Comment:
This code can be improved.  You have 6 lines repeated in both branches of the if. Also it is not necessary to have the temporary argString content array. `instruction` is not a good name: should be something plural like insnWords or words.

Reply:
Fixed